samples/csharp/getting-started/Ranking_PersonalizedSort/README.md

```diff
@@ -0,0 +1,215 @@
+
+## Dataset
+The training and testing data used by this sample is based on a public [dataset available at Kaggle](https://www.kaggle.com/c/expedia-personalized-sort) originally provided by Expedia (https://www.expedia.com).
+
+Expedia's datasets consist of hotel search results that are grouped according to a user's query; each hotel result includes the following details:
+* Hotel attributes, such as location attractiveness and price.
+* User's criteria for searching hotels, such as the number of rooms\children\adults, length of stay, etc.
+* User's purchase and browsing history, such as whether they clicked the link of a hotel or purchased\booked it.
+* Information on similar competitor hotel offerings.
+
+## ML Task - Ranking
+As previously mentioned, this sample uses the LightGbm Lambdarank algorithm which is applied using a supervised learning technique known as "Learning to Rank".  This technique requires that train/test datasets contain groups of data instances that are labeled with their ideal ranking value.  The label is a numerical\ordinal value, such as {4, 3, 2, 1, 0} or a text value {"Perfect", "Excellent", "Good", "Fair", or "Bad"}.  The process for labeling these data instances with their ideal ranking value can be done manually by subject matter experts.  Or, the labels can be determined using other metrics, such as the number of clicks on a given search result. This sample uses the latter approach.
+
+Once the train/test datasets are labeled with ideal ranking values, the model (e.g. ranker) can then be trained and tested using this data.  Through the model training process, the ranker learns how to score each data instance within a group based on their label value.  The resulting score of an individual data instance by itself isn't important -- instead, the scores should be compared against one another to determine the relative ordering of a group's data instances.  The higher the score a data instance has, the more relevant and more highly ranked it is within its group.      
+
+## Solution
+The sample performs the following high-level steps to rank Expedia hotel search results:
+1. Each hotel search result is **labeled** with its ideal ranking value.
+2. Once the dataset is labeled, the data is **split** into training and testing datasets.  
+3. The model is **trained** using the train dataset using LightGbm Lambdarank algorithm.  
+4. The model is **tested** using the test dataset.  This results in a **prediction** that includes a **score** for each hotel instance.  The score is used to determine the ranking relative to other hotels within the same query (e.g. group).  The predictions are then **evaluated** by examining metrics; specifically the [Discounted Cumulative Gain](https://en.wikipedia.org/wiki/Discounted_cumulative_gain).
```

Comment:
I'd focus on the NDCG metric as its comparable across datasets, whereas DCG heavily depends on the dataset:
```suggestion
4. The model is **tested** using the test dataset.  This results in a **prediction** that includes a **score** for each hotel instance.  The score is used to determine the ranking relative to other hotels within the same query (e.g. group).  The predictions are then **evaluated** by examining metrics; specifically the [Normalized Discounted Cumulative Gain](https://en.wikipedia.org/wiki/Discounted_cumulative_gain) (NDCG).
```